Automattic/wp-calypso
# Gutenberg: Update Publicize to use social-icons directly

This is an experiment to tackle #28096 

Cons: it adds the full `social-icons` build bundle (over 100K), so I'm not sure if we want to do it this way. Instead, we might want to have `social-icons` build standalone versions of the icons that we could use without bundling the entire library.

#### Changes proposed in this Pull Request

* Update Publicize to use `social-icons` for the social icons.
* Update Publicize to use Gutenberg's `Dashicon` component for the plus icon.

#### Testing instructions

* Spin up a site with this branch - https://jurassic.ninja/create?shortlived&gutenpack&gutenberg&calypsobranch=update/gutenberg-publicize-use-social-icons&branch=try/publicize-gutenberg-block-externally-built-rebased
* Uncomment this line: https://github.com/Automattic/wp-calypso/blob/f68ef27219b5818284f08d8f41f3865bd3222fc7/client/gutenberg/extensions/presets/jetpack/editor.js#L9
* Write a new post.
* Attempt to publish the post.
* Create a connection for each connection type.
* Verify icons still look well.
* Verify connection add button icons still looks well.

Fixes #28096
